src/App.tsx

```diff
@@ -1,34 +1,72 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { actions as todosActions } from './features/todos';
+import { ErrorModal } from './components/ErrorModal';
 
 export const App: React.FC = () => {
+  const [isLoader, setIsLoader] = useState(false);
+  const [errorTitle, setErrorTitle] = useState('');
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const todosData = useAppSelector(state => state.todos);
+  const dispatch = useAppDispatch();
+  const set = (data: Todo[]) => dispatch(todosActions.setTodos(data));
+
+  const loadData = async () => {
+    try {
```

Comment:
jfyi - considered a best practice to move complex logic such as API calls out of the component. Consider creating a separate action for loading todos in the todos slice and handle the API call there using a middleware like redux-thunk.

Reply:
i think it`s in the next lesson